src/index.js

```diff
@@ -370,6 +373,7 @@ function removeNodeFromRoot (root, node) {
 function removeSlotFromRoot (root, node) {
   node.assignedNodes().forEach(slotNodeFromSlot);
   delete rootToSlotMap.get(root)[getSlotNameFromSlot(node)];
+  delete slotToRootMap.get(node);
```

Comment:
This won't work as you're not deleting a key. You can call `slotToRootMap.delete(node)`, though.